internal/monitoring/auto_tags.go

```diff
@@ -1,53 +1,48 @@
 package monitoring
 
 import (
-	"encoding/json"
+	"github.com/keptn-contrib/dynatrace-service/internal/dynatrace"
 	"github.com/keptn-contrib/dynatrace-service/internal/lib"
 
 	log "github.com/sirupsen/logrus"
 )
 
 type AutoTagCreation struct {
-	client *lib.DynatraceHelper
+	client *dynatrace.DynatraceHelper
 }
 
-func NewAutoTagCreation(client *lib.DynatraceHelper) *AutoTagCreation {
+func NewAutoTagCreation(client *dynatrace.DynatraceHelper) *AutoTagCreation {
 	return &AutoTagCreation{
 		client: client,
 	}
 }
 
 // Create creates auto-tags in Dynatrace and returns the tagging rules
-func (at *AutoTagCreation) Create() []lib.ConfigResult {
-	var taggingRules []lib.ConfigResult
+func (at *AutoTagCreation) Create() []dynatrace.ConfigResult {
+	var taggingRules []dynatrace.ConfigResult
 	if !lib.IsTaggingRulesGenerationEnabled() {
 		return taggingRules
 	}
 
 	log.Info("Setting up auto-tagging rules in Dynatrace Tenant")
 
-	response, err := at.client.SendDynatraceAPIRequest("/api/config/v1/autoTags", "GET", nil)
+	autoTagsClient := dynatrace.NewAutoTagClient(at.client)
+	existingDTRules, err := autoTagsClient.Get()
 	if err != nil {
 		// Error occurred but continue
-		log.WithError(err).Error("Could not get existing tagging rules")
-	}
-
-	existingDTRules := &lib.DTAPIListResponse{}
-	err = json.Unmarshal([]byte(response), existingDTRules)
-	if err != nil {
-		// Error occurred but continue
-		log.WithError(err).Error("Failed to unmarshal Dynatrace tagging rules")
+		// TODO 2021-08-18: should this error just be ignored?
+		log.WithError(err).Error("Failed retrieving Dynatrace tagging rules")
```

Comment:
I would propose to resolve this todo by generating an error condition if possible.

Reply:
This would change the flow of logic, so we will need to focus on this (and other todos) as well as error handling in general in the near future. We will create issues for that